src/plugins/kibana_react/public/validated_range/validated_dual_range.tsx

```diff
@@ -18,17 +18,38 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { ReactNodeLike, Requireable } from 'prop-types';
+import { EuiFormRow, EuiDualRange, EuiRangeProps } from '@elastic/eui';
+import { EuiFormRowDisplayKeys } from '@elastic/eui/src/components/form/form_row/form_row';
 import { isRangeValid } from './is_range_valid';
 
-import { EuiFormRow, EuiDualRange } from '@elastic/eui';
-
 // Wrapper around EuiDualRange that ensures onChange callback is only called when range value
 // is valid and within min/max
-export class ValidatedDualRange extends Component {
-  state = {};
 
-  static getDerivedStateFromProps(nextProps, prevState) {
+interface ValidatedRangeValues extends Omit<EuiRangeProps, 'value'> {
+  value?: [number | string, number | string] | undefined;
```

Comment:
You named `number | string` in `isRangeValid` as `RangeValue`. Lets use that type here. EuiDualRangeProps has already created this type as well its called `ValueMember`. Lets use that naming to be consistent and put a TODO linking to a github issue in EUI to expose this type. Then we can just use the type from EUI in the future.

Reply:
Opened an issue [here](https://github.com/elastic/eui/issues/3054) and got the response to use `EuiDualRangeProps['value'][0]` which seemed to work well. Let me know if that works for you and I'll close the issue